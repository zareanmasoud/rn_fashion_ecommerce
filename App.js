/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React from 'react';
import NavigationContainer from './src/screens/routes/container';
import {ScreenContentFrame} from './src/lib/components/ScreenContentFrame';

const App: () => React$Node = () => {
  return (
    <ScreenContentFrame>
      <NavigationContainer/>
    </ScreenContentFrame>
  );
};

export default App;

