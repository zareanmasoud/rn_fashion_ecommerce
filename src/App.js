/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {useState} from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import NavigationContainer from 'screens/routes/container';
// eslint-disable-next-line import/no-extraneous-dependencies
// import ScreenContentFrame from 'lib/components/ScreenContentFrame';
import {CartProvider} from './context/CartContext';

export default function App() {
  const [cart, setCart] = useState(0);

  const addToCart = () => {
    setCart(cart + 1);
  };

  const removeFromCart = () => {
    setCart(cart - 1);
  };
  return (
    // <ScreenContentFrame>
    <CartProvider value={{cart, addToCart, removeFromCart}}>
      <NavigationContainer />
    </CartProvider>
    // </ScreenContentFrame>
  );
}
