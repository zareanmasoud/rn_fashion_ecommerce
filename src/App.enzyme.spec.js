/**
 * @jest-environment jsdom
 */
import React from 'react';
import {shallow} from 'enzyme';
import App from './App';

describe('App tested with airbnb enzyme', () => {
  test('App mount with enzyme', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
