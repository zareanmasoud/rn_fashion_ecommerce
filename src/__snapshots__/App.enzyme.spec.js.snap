// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tested with airbnb enzyme App mount with enzyme 1`] = `
<ScreenContentFrame
  statusBackgroundColor="#ffffff"
  statusBarStyle="dark-content"
>
  <NavigationContainer
    theme="light"
  />
</ScreenContentFrame>
`;
